### DISCUSSION TITLE
Cooja button click

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2668] [General]
This is a very minor item. There are times when clicking the Start/Pause button in Cooja where it would not start. The message in terminal will be. Possibly a debouncer UI issue.

INFO [sim] (Simulation.java:472) - Runtime: 0 ms. Simulated time: 0 ms. Speedup: 0.0

**Comment by pjonsson** [comment]
I don't really use Cooja, but I have never had that happen to me. Can you try the latest Cooja on develop directly on your Mac (without virtualization) and see if it happens there too?

**Reply by rahav** [reply]
This occurs when Cooja runs in a Contiki container on Ubuntu 22.04.

My Mac has Apple CPU and I was not able to make Contiki run on it. If you have any advise, I'll be happy to install it.

I have an older Mac with Intel CPU that I can recommission. If that's what you meant, which way to do want me to install it? (a link to the directions will suffice).

**Reply by pjonsson** [reply]
It didn't make it to 4.9, but develop has the patches for M1/M2 merged, so everything should work out of the box there, as long as you have gawk and GNU make 4 from Homebrew (or other package repository) first in your PATH.

My suggestion isn't that you stop using whatever you are currently using for your production, I just want to know if you can reproduce the issue without all the intermediate layers.

**Reply by rahav** [reply]
Indeed, this is how I understood your answer and was trying to help by running it on another OS. This does occur without virtualization layers, when I run it on directly on Ubuntu. This is not important or time sensitive by any means, because I did not see it add anything to the LOG files, so it is very likely just cosmetic.

To replicate: Start Cooja UI and run a few simulations without exiting Cooja. I could not find any pattern, but after a few cycles of running a few simulations by clicking `Reload` and `Start`, the `Start/Pause` button will not respond unless it is clicked again.

**Reply by pjonsson** [reply]
This was hard to reproduce, which makes me think it depends on the speed of your computer (and whether you have ccache installed). My guess is that there is some kind of race between the shutdown and opening of a new simulation, and how events arrive on the AWT thread. The issue should obviously be fixed, but it will be hard to identify where the short window with the race condition is, and even if we find it, it might still be difficult to fix.

You might be able to avoid the problem if you wait a little before you click start after reloading (but I personally hate when programs force me to slow down to avoid hitting bugs).

**Reply by rahav** [reply]
Thank you for looking into this and the reply. From your analysis it seems to not affect the simulation itself, which was my main concern. I will use the workaround you suggested.

**Reply by pjonsson** [reply]
Can you reproduce the issue if instead of clicking, you reload with Ctrl-R and start/stop with Ctrl-S?

**Reply by rahav** [reply]
I will try that.

**Reply by rahav** [reply]
After running numerous simulations, the problem did not occur when using the keyboard shortcuts.

For reference, the workflow we follow is:
Open a `.csc` file. Then the following cycle will be followed:
Append to file `x` --> Run the sim --> Stop appending --> Change the configuration in an external file --> Reload the sim --> ...

After a few such cycles we will open a different a `.csc` file and go through above cycle.

Because of the manual steps, it should not be a race condition due to speed.
